.github/workflows/test-env-cleanup.yml

```
@@ -8,6 +8,9 @@ on:
 jobs:
   cleanup:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write # needed by aws-actions/configure-aws-credentials
+      contents: read
```

Missing `deployments: write` (step with the `bobheadxi/deployments` action)

```suggestion
      contents: read
      deployments: write
```